### PR TITLE
DROOLS-6268 : Fix for GRE in

### PR DESCRIPTION
**Pull request by Rikkola**
**JIRA**:  

[DROOLS-6268](https://issues.redhat.com/browse/DROOLS-6268)

**referenced Pull Requests**: 

* https://github.com/kiegroup/kie-soup/pull/193
* https://github.com/kiegroup/drools/pull/3524
* https://github.com/kiegroup/kie-wb-common/pull/3619
* https://github.com/kiegroup/drools-wb/pull/1481

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

File: drools-workbench-models/drools-workbench-models-commons/src/main/java/org/drools/workbench/models/commons/backend/rule/RuleModelDRLPersistenceImpl.java [hunk 1]
```diff
@@ -4365,7 +4373,8 @@ public FieldConstraint asFieldConstraint(final RuleModel m,
          * Fact(something != null && matches "P.*") = was supported before, now is deprecated.
          * It breaks the UI if user has a rule with this case, so we convert to the new format:
          * Fact(something != null && something matches "P.*")
-         *                           ^^^^^^^^
+         * ^^^^^^^^
+         *
```
Comment (Rikkola) [inline drools-workbench-models/drools-workbench-models-commons/src/main/java/org/drools/workbench/models/commons/backend/rule/RuleModelDRLPersistenceImpl.java]: Need to fix this.

**Review by karreiro — APPROVED** [review]
Thank you, @Rikkola!

File: drools-workbench-models/drools-workbench-models-commons/src/main/java/org/drools/workbench/models/commons/backend/rule/RuleModelDRLPersistenceImpl.java [hunk 2]
```diff
@@ -1441,7 +1441,12 @@ private void buildEnumFieldValue(final String operator,
         protected void buildDefaultFieldValue(final String operator,
                                               final String value,
                                               final StringBuilder buf) {
-            if (!operator.equals("== null") && !operator.equals("!= null")) {
+
+            if (operator.equals("in") || operator.equals("not in")) {
+                buf.append(" (");
+                buf.append(value);
+                buf.append(")");
+            } else if (!operator.equals("== null") && !operator.equals("!= null")) {
```
Comment (jomarko) [inline drools-workbench-models/drools-workbench-models-commons/src/main/java/org/drools/workbench/models/commons/backend/rule/RuleModelDRLPersistenceImpl.java]: Could we please use:
- `Objects.equals(operator, in)`, same for "not in"
- `Objects.equals(operator, "== null")`, same for "!= null"

It is a preventing from NPE if `operator` is null.

**Review by jomarko — COMMENTED** [review]
Code review is inline.

**Review by jomarko — APPROVED** [review]

**Comment by Rikkola** [comment]
Jenkins run fdb

**Comment by Rikkola** [comment]
`[2021-04-15T07:00:34.424Z] [ERROR]  You have triggered an abuse detection mechanism. Please wait a few minutes before you try again.`

Jenkins run fdb

**Comment by Rikkola** [comment]
Jenkins run fdb

**Comment by Rikkola** [comment]
Jenkins retest this

**Comment by Rikkola** [comment]
Jenkins retest this

**Comment by Rikkola** [comment]
```
org.kie.workbench.common.widgets.client.datamodel.AsyncPackageDataModelOracleImplTest.testGetOperatorCompletionsForThis[0]
org.kie.workbench.common.widgets.client.datamodel.AsyncPackageDataModelOracleImplTest.testGetOperatorCompletionsForBoolean[0]
org.kie.workbench.common.widgets.client.datamodel.AsyncPackageDataModelOracleImplTest.testGetOperatorCompletionsForString[0]
org.kie.workbench.common.widgets.client.datamodel.AsyncPackageDataModelOracleImplTest.testGetOperatorCompletionsForInteger[0]
org.kie.workbench.common.widgets.client.datamodel.AsyncPackageDataModelOracleImplTest.testGetOperatorCompletionsForThis[1]
org.kie.workbench.common.widgets.client.datamodel.AsyncPackageDataModelOracleImplTest.testGetOperatorCompletionsForBoolean[1]
org.kie.workbench.common.widgets.client.datamodel.AsyncPackageDataModelOracleImplTest.testGetOperatorCompletionsForString[1]
org.kie.workbench.common.widgets.client.datamodel.AsyncPackageDataModelOracleImplTest.testGetOperatorCompletionsForInteger[1]
org.kie.workbench.common.widgets.client.datamodel.PackageDataModelOracleCEPCompletionsTest.testCEPCompletions
org.kie.workbench.common.widgets.client.datamodel.PackageDataModelOracleCompletionsTest.testFactFieldConnectiveOperatorCompletions
```

These are actually issues I need to fix and make a new PR for kie-wb-common.

**Comment by Rikkola** [comment]
Jenkins run fdb

**Comment by Rikkola** [comment]
Jenkins run fdb

**Comment by Rikkola** [comment]
Jenkins retest this

**Comment by Rikkola** [comment]
Jenkins run fdb

**Comment by Rikkola** [comment]
Jenkins retest this

**Comment by Rikkola** [comment]
Jenkins run fdb

**Comment by Rikkola** [comment]
`org.jbpm.executor.impl.jms.JmsAvaiableJobExecutorTest.testAsyncAuditErrorProducer`

jenkins run fdb

**Comment by Rikkola** [comment]
`Setting status of d7f53fa20b2b2a79d6302d1b01e994d943f99ea2 to FAILURE with url https://eng-jenkins-csb-business-automation.apps.ocp4.prod.psi.redhat.com/job/KIE/job/master/job/fdb/job/drools-master.fdb/30/ and message: 'Build finished. 35815 tests run, 581 skipped, 0 failed.'`
jenkins run fdb

**Comment by jomarko** [comment]
@Rikkola I do not understand our ensemble is not green and others are getting in. The only idea I have, our ensemble is not rebased with current master, could that help?

**Comment by Rikkola** [comment]
@jomarko worth trying at this point.

**Comment by Rikkola** [comment]
jenkins run fdb

**Comment by Rikkola** [comment]
jenkins run fdb

**Comment by jomarko** [comment]
@Rikkola one of fails is not random, see https://issues.redhat.com/browse/DROOLS-6436

**Comment by Rikkola** [comment]
@jomarko ok thanks. I see that one failed in the other PR.

**Comment by Rikkola** [comment]
jenkins run fdb

**Comment by jomarko** [comment]
@Rikkola I thought we agreed in chat to merge?

**Comment by jomarko** [comment]
I checked the failing test `LocalStorageStreamingKieSessionTest` locally and it is passing. The issue is probably we run in in parallel on jenkins?

Also `ContainerRemoteStatusPresenterTest.testOnServerInstanceUpdatedNewInstance` shouldn't  be failing anymore as we merged https://issues.redhat.com/browse/DROOLS-6436. Could we please try to rebase this PR?

**Pull request merged**